src/passes/OptimizeInstructions.cpp

```diff
@@ -1663,6 +1663,16 @@ struct OptimizeInstructions
         }
       }
     }
+    {
+      // x * 2.0  ==>  x + x
```

Comment:
It would be good to add a comment explaining that we only do this when it doesn't require adding a new local.

Reply:
Sure!